### PR TITLE
Fix verifying certs on windows

### LINKED ISSUE #22: Certs do not verify when running on windows
This issue was discussed in https://github.com/vmware-tanzu/carvel-kbld/issues/75.

Since `imgpkg` inherited some functionality from `kbld` we will need to fix it in `imgpkg` as well.

Notes
- see [kbld's comment](https://github.com/vmware-tanzu/carvel-kbld/issues/75#issuecomment-906670951) about reproducing this again before starting implementation.

### PR DESCRIPTION
**Pull request by DennisDenuto**
Attempts to fix #22 

TODO:

- [x] document how to use custom ca certs with windows root store
- [x] figure out why windows still has a file handle on the bundle.tar file. TODO in test code
- [x] confirm ca certs work in an airgapped env
- [x] investigate if theres any performance cost due to using windows root ca store (I think it downloads ca certs using network)

Authored-by: Dennis Leon <leonde@vmware.com>

### CODE REVIEW

File: pkg/imgpkg/cmd/push_test.go [hunk 1]
```diff
@@ -56,7 +57,7 @@ func TestMultiImgpkgDirError(t *testing.T) {
 		t.Fatalf("Expected validations to err, but did not")
 	}
 
-	if !strings.Contains(err.Error(), "This directory contains multiple bundle definitions. Only a single instance of .imgpkg/images.yml can be provided") {
+	if !strings.Contains(err.Error(), fmt.Sprintf("This directory contains multiple bundle definitions. Only a single instance of .imgpkg%simages.yml can be provided", string(os.PathSeparator))) {
```
Comment (cppforlife) [inline pkg/imgpkg/cmd/push_test.go:60]: im not sure how i feel about us flipping slashes in error messages. should we do this or just leave it as is?

Comment (DennisDenuto) [reply pkg/imgpkg/cmd/push_test.go:60]: hmmm i'm not sure i fully understand here...

This is within a test asserting an error message though... so the implementation on windows is going to return a different error message when running this test in linux (the diff being the slash). So this aims to make the test make the right assertion based on the OS that is running said test.

Comment (cppforlife) [reply pkg/imgpkg/cmd/push_test.go:60]: i see, we are generating that stuff on the fly.

File: pkg/imgpkg/registry/registry.go [hunk 2]
```diff
@@ -250,8 +251,17 @@ func (r Registry) FirstImageExists(digests []string) (string, error) {
 }
 
 func newHTTPTransport(opts Opts) (*http.Transport, error) {
-	pool, err := x509.SystemCertPool()
-	if err != nil {
+	var pool *x509.CertPool
+
+	if runtime.GOOS != "windows" {
```
Comment (cppforlife) [inline pkg/imgpkg/registry/registry.go:259]: lets throw in a comment here why we are doing this.

Comment (DennisDenuto) [reply pkg/imgpkg/registry/registry.go:259]: done

File: test/helpers/assets.go [hunk 3]
```diff
@@ -94,6 +94,7 @@ func (a *Assets) CleanCreatedFolders() {
 	a.T.Helper()
 	for _, folder := range a.CreatedFolders {
 		err := os.RemoveAll(folder)
+
```
Comment (cppforlife) [inline test/helpers/assets.go]: nit: remove newline?

File: hack/build.sh [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 set -e -x -u
 
+FILE_EXT="${1-}"
```
Comment (cppforlife) [inline hack/build.sh]: instead of turning this into an arg wdyt about just using $IMGPKG_BINARY_EXT env var

File: .github/workflows/gh-test.yml [hunk 5]
```diff
@@ -28,4 +28,36 @@ jobs:
...
+        bash -c '
+        set -e -x
+
+        export GOPATH=$(echo `pwd`)
+        export PATH="$GOPATH/bin:$PATH"
+        cd src/github.com/${{ github.repository }}
+
+        export IMGPKG_E2E_IMAGE="ttl.sh/github-action-test-relocation-$GITHUB_RUN_ID"
+        export IMGPKG_E2E_RELOCATION_REPO="ttl.sh/github-action-imgpkg-test-$GITHUB_RUN_ID"
+        # deploy local registry and run tests
```
Comment (cppforlife) [inline .github/workflows/gh-test.yml]: this comment doesnt seem to belong to anything

File: test/e2e/gcloud_expired_token_test.go [hunk 6]
```diff
@@ -9,13 +9,17 @@ import (
...
 
 	"github.com/k14s/imgpkg/test/helpers"
 	"github.com/stretchr/testify/require"
 )
 
 func TestCopyWithBundleLockInputToRepoDestinationUsingGCloudWithAnExpiredToken(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipping test as docker image used requires linux")
+	}
 	env := helpers.BuildEnv(t)
```
Comment (cppforlife) [inline test/e2e/gcloud_expired_token_test.go:24]: nit: seems like in other places you gotta newline after this windows if

File: test/e2e/copy_from_image_test.go [hunk 7]
```diff
@@ -116,7 +113,7 @@ func TestCopyAnImageFromATarToARepoThatDoesNotContainNonDistributableLayersButTh
 		// copy to tar skipping NDL
 		imgpkg.Run([]string{"copy", "-i", airgappedRepo, "--to-tar", tarFilePath})
 
-		stopRegistryForAirgapTesting(t, env)
+		stopRegistryForAirgapTesting(fakeRegistry)
```
Comment (cppforlife) [inline test/e2e/copy_from_image_test.go:116]: should this function be replaced with defer fakeRegistry.CleanUp() after startRegistryForAirgapTesting()?

Comment (cppforlife) [reply test/e2e/copy_from_image_test.go:116]: oh i see what's going on you are clearing it before copy... nm

File: .github/workflows/gh-test.yml (hunk 5 above)
Comment (DennisDenuto) [reply .github/workflows/gh-test.yml]: true. removed.

File: hack/build.sh (hunk 4 above)
Comment (DennisDenuto) [reply hack/build.sh]: done

File: test/helpers/assets.go (hunk 3 above)
Comment (DennisDenuto) [reply test/helpers/assets.go]: done

**Pull request merged**